### DISCUSSION TITLE
Test Driver temporary files in bin folder

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#13597] [General]
Hi, when creating tests using the TestDriver, it creates topology and other files in the `bin` folder. 

Those files are not deleted and as times go by it adds up to space consumption.

Example: 
- *topology files
- RavenDB folder
- RavenDBServer folder

I had for me the RavenDB and RavenDBServer folder grow up to 30GB for a project after some time.

Would it make sense that the TestDriver cleans up all temporary files created after the test finishes?

**Comment by ayende** [comment]
Yes, that is something that we can do: https://issues.hibernatingrhinos.com/issue/RavenDB-17980

**Reply by mcrio** [reply]
Thanks Oren, sounds good. So the main disk space consumers seem to be the folders "RavenDB" and "RavenDBServer".

**Reply by ayende** [reply]
Those are the actual _databases_ in use.
Are you disposing the instances? Because that should delete them

**Reply by mcrio** [reply]
Thanks Oren. I see now. My bad, I was not disposing it. It does delete the temporary files in the Server folder.

**Reply by ppekrol** [reply]
You can also disable topology cache (RavenDB 5.2+) by overriding PreInitialize

e.g.

```
        protected override void PreInitialize(IDocumentStore documentStore)
        {
            documentStore.Conventions.DisableTopologyCache = true;
        }
```

**Reply by mcrio** [reply]
Thanks @ppekrol this stops producing the database-topology files but the cluster-topology files still remain.
